### DISCUSSION TITLE
How to replace some strings like `__(foo + 'bar')`  with some strings via webpack plugin

### DISCUSSION THREAD

**Discussion by liuliangsir** [webpack/webpack#15241] [Q&A] [answered]
I am going to use some patterns like `__('Hello' + foo)` for i18n template.  Here are some steps:

- write some codes like `const content = __('hello' + foo)` in a js/ts file
- replace the string above with translated content like `const content = foo + '您好'` via webpack plugin during compilation stage

I have tried some solutions like `toConstantDependency` in webpack plugin. Unfortunately, it seems not work well with some strings, which contains dynamic content such as variable.

**Comment by liuliangsir** [comment]
@alexander-akait cc

**Comment by vankop** [comment]
Did not get what you want to do.. Could you describe step by step ?

**Comment by liuliangsir** [comment]
@vankop I've updated the problem description

**Comment by vankop** [comment, marked as answer]
I think `RuntimeValue` in `DefinePlugin` will be fine for you https://github.com/webpack/webpack/blob/main/test/watchCases/plugins/define-plugin/webpack.config.js#L10

**Reply by vankop** [reply]
https://webpack.js.org/plugins/define-plugin/#runtime-values-via-runtimevalue

**Reply by vankop** [reply]
or you what to execute i18n template in runtime? I mean do you need build time value or runtime value?

**Reply by liuliangsir** [reply]
build time value

**Reply by vankop** [reply]
so `RuntimeValue` in `DefinePlugin` should be ok for you
